synapse/rest/client/v1/room.py

```diff
@@ -171,6 +172,77 @@ def on_PUT(self, request, room_id, event_type, state_key, txn_id=None):
         if state_key is not None:
             event_dict["state_key"] = state_key
 
+        # If the event is a retention policy definition for the room, check that it
+        # respects the boundaries imposed by the server's administrator.
+        if event_type == EventTypes.Retention:
+            min_lifetime = content.get("min_lifetime")
+            max_lifetime = content.get("max_lifetime")
+
+            if min_lifetime is not None:
+                if (
+                    self.config.retention_allowed_lifetime_min is not None
+                    and min_lifetime < self.config.retention_allowed_lifetime_min
```

Comment:
we should probably check that it is a valid `int` rather than explode with a 500?